src/character.cpp

```diff
@@ -1032,7 +1032,13 @@ void Character::set_max_power_level( units::energy npower_max )
 
 void Character::mod_power_level( units::energy npower )
 {
-    power_level = clamp( power_level + npower, 0_kJ, max_power_level );
+    int new_power = units::to_millijoule( power_level ) + units::to_millijoule( npower );
+    // An integer overflow has occurred - the sum of two positive things (power_level is always positive)
+    // cannot be negative, so if it is, we know integer overflow has occured
+    if( npower >= 0_mJ && new_power < 0 ) {
```

Comment:
This test is relying on signed integer overflow, which is undefined behaviour, so it could potentially break depending on compiler optimization.  Easiest solution is to do the computation in `int64_t`s instead, then clamp and cast back to `int`.  Or you can do the more roundabout approach of subtracting `npower` from `max_power_level` and comparing with `power_level` (but then you start risking underflow if you're not careful, so it's hard to do correctly).

Reply:
Whoops, thanks #36686!